includes/Story_Post_Type.php

```diff
@@ -694,6 +695,27 @@ public function filter_template_include( $template ) {
 		return $template;
 	}
 
+	/**
+	 * Redirect old url slug. /stories -> /web-stories.
+	 *
+	 * @return void
+	 */
+	public function redirect_old_slug() {
```

Comment:
My main worry here is that we wrongly redirect `/stories` to `/web-stories` when `/stories` is a legitimate static page in WordPress.

We want to avoid such conflicts, which is the main reason for changing the slug after all.

Would `redirect_canonical` be a more suitable filter for this?

Reply:
I have added a check for `is_404`. Only redirect if 404, so that should fix that. 

I can move the filter, if you want, but this works.

Reply:
The `redirect_canonical` filter seems more fitting I think.

It already nicely parses `$_SERVER['REQUEST_URI'] and passes it to the filter, and handles the  redirect for us.

This should make it easier to test as well.

Reply:
I looking into and `redirect_canonical` just hook into `template_redirect`. [See](https://github.com/WordPress/wordpress-develop/blob/db9a91a2a88595aa412d8b3905187a8b2c7d2147/src/wp-includes/default-filters.php#L528).

I am not sure I see the point of just hook in a function that already hooks in here. I like this hook gives me more control. 

I also tried to get it working locally and it didn't work for some reason. I think maybe something to do with multisite...

Reply:
Easier testing would be one reason.

Speaking of, can you add tests for this?

Reply:
There are now tests.